### PR TITLE
feat: add application confirmation modals (#4154)

### PR DESCRIPTION
**Pull request by emilyjablonski**
Pulls over the following commit from core:
https://github.com/bloom-housing/bloom/pull/4154

### CODE REVIEW

**Review by mcgarrye — COMMENTED** [review]
Testing looks good, just wondering why some file changes weren't brought over from core?

Screenshot 2024-07-03 at 11 21 11 AM

File: sites/partners/src/pages/listings/[id]/applications/index.tsx [hunk 1]
```diff
@@ -177,7 +181,15 @@ const ApplicationsList = () => {
                 headerContent={
                   <div className="flex gap-2 items-center">
                     <Button
-                      href={`/listings/${listingId}/applications/add`}
+                      onClick={() => {
+                        if (listingDto.lotteryLastRunAt) {
```
Comment (mcgarrye) [inline sites/partners/src/pages/listings/[id]/applications/index.tsx:185]: Was playing around a bit more and noticed that while this should never be set with the flag turned off, this is not feature flagged

Comment (emilyjablonski) [reply sites/partners/src/pages/listings/[id]/applications/index.tsx:185]: Was relying on the fact that this cannot be set without the feature being on - but am flexible here, would want to make that a core change if preferred

**Comment by emilyjablonski** [comment]
@mcgarrye If you look at the differences, the changes in core were upgrading the seeds version, and that version already existed in HBA from a different PR

File: sites/partners/src/pages/listings/[id]/applications/index.tsx (hunk 1 above)
Comment (mcgarrye) [reply sites/partners/src/pages/listings/[id]/applications/index.tsx:185]: I think it's fine as is. Just wanted to call it out to make sure it was intentional

**Review by mcgarrye — APPROVED** [review]
LGTM

File: sites/partners/src/pages/listings/[id]/applications/index.tsx (hunk 1 above)
Comment (emilyjablonski) [reply sites/partners/src/pages/listings/[id]/applications/index.tsx:185]: I'll pick this up in my next core lottery PR!

**Pull request merged**